oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java

```diff
@@ -75,10 +86,87 @@ public static <T> Set<T> toSet(final Iterable<T> iterable) {
      * @return the set
      * @param <T> the type of the elements
      */
-    public static <T> Set<T> toSet(final Iterator<T> iterator) {
+    @NotNull
+    public static <T> Set<T> toSet(@NotNull final Iterator<T> iterator) {
+        Objects.requireNonNull(iterator);
         final Set<T> result = new HashSet<>();
         iterator.forEachRemaining(result::add);
         return result;
     }
 
-}
+    /**
+     * Convert a vararg list of items to a set.  The returning set is mutable and supports all optional operations.
+     * @param elements elements to convert
+     * @return the set
+     * @param <T> the type of the elements
+     */
+    @SafeVarargs
+    @NotNull
+    public static <T> Set<T> toSet(@NotNull final T... elements) {
+        Objects.requireNonNull(elements);
+        // make sure the set does not need to be resized given the initial content
+        float loadFactor = (float) 0.75; // HashSet default
+        int initialCapacity = 1+ (int) (elements.length / loadFactor);
+        final Set<T> result = new HashSet<>(initialCapacity, loadFactor);
+        for (T element : elements) {
+            result.add(element);
+        }
+        return result;
+    }
+
+    /**
+     * Convert an {@code Iterator} to an {@code Iterable}.
+     *
+     * @param iterator
+     *            iterator to convert
+     * @return a single-use iterable for the iterator (representing the remaining
+     *         elements in the iterator)
+     * @throws IllegalStateException
+     *             when {@linkplain Iterable#iterator()} is called more than
+     *             once
+     */
+    @NotNull
+    public static <T> Iterable<T> toIterable(@NotNull final Iterator<T> iterator) {
+        Objects.requireNonNull(iterator);
+
+        Iterable<T> delegate = new Iterable<T>() {
+
+            private boolean consumed = false;
+
+            @Override
+            public Iterator<T> iterator() {
+                if (consumed) {
+                    throw new IllegalStateException("Iterator already returned once");
+                } else {
+                    consumed = true;
+                    return iterator;
+                }
+            }
+        };
```

Comment:
This is not thread safe, it can be consumed two or more times under concurrent access. If we don't want to make it thread safe, would be better to mention it in the Java docs.

Reply:
will do